cmd/cluster-display/main.go

```diff
@@ -161,13 +162,62 @@ func (g *clusterInfoGetter) GetClusterDetails(ctx context.Context, cluster strin
 	}, nil
 }
 
-func getClusterPage(ctx context.Context, clients map[string]ctrlruntimeclient.Client, skipHive bool, getter ClusterInfoGetter) *Page {
+type memoryCache struct {
+	clusterDataMutex         sync.Mutex
+	clusterData              []map[string]string
+	clusterDataLastUpdatedAt time.Time
+
+	ClusterPoolDataMutex         sync.Mutex
+	ClusterPoolData              []map[string]string
+	ClusterPoolDataLastUpdatedAt time.Time
+}
+
+func (c *memoryCache) GetClusterPoolPage(ctx context.Context, client ctrlruntimeclient.Client) (*Page, error) {
+	c.ClusterPoolDataMutex.Lock()
+	defer c.ClusterPoolDataMutex.Unlock()
+	if c.ClusterPoolData == nil || time.Now().After(c.ClusterPoolDataLastUpdatedAt.Add(time.Hour)) {
+		data, err := getClusterPool(ctx, client)
+		if err != nil {
```

Comment:
We return an error here, but some of the value I was thinking we could achieve by adding the cache is to continue to return the previous results in this state. That way we could still display the last available information about the cluster. We could maybe display a warning that there is a problem, and one or more cluster may be unreachable or outdated along with this info.

Reply:
> We could maybe display a warning that there is a problem

Added a log in the backend instead.